pkg/backend/diy/postgres/bucket.go

```diff
@@ -0,0 +1,509 @@
+// Copyright 2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package postgres implements a blob.Bucket storage backend using PostgreSQL.
+//
+// SECURITY NOTE - SQL Injection Prevention:
+// This implementation uses dynamic table names in SQL queries, which could appear
+// to be vulnerable to SQL injection. However, this is safe because:
+//
+//  1. Table names come from the PostgreSQL connection string configuration,
+//     which is controlled by system administrators/developers, not end users
+//  2. The connection string is part of the backend configuration, not user input
+//  3. All actual user-provided data (keys, blob data) are properly parameterized
+//     using prepared statement placeholders ($1, $2, etc.)
+//  4. This pattern allows for configurable table schemas while maintaining security
+//
+// The //nolint:gosec comments acknowledge that static analysis tools may flag
+// these patterns, but they are intentionally safe in this controlled context.
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+
+	_ "embed"
+
+	"gocloud.dev/blob"
+	"gocloud.dev/blob/driver"
+	"gocloud.dev/gcerrors"
+
+	// Import the PostgreSQL driver
+	_ "github.com/lib/pq"
+)
+
+// blobData represents the JSON structure for storing blob data in PostgreSQL
+type blobData struct {
+	Data string `json:"data"` // base64 encoded binary data
+}
+
+// Bucket implements blob.Bucket storage using PostgreSQL.
+type Bucket struct {
+	db        *sql.DB
+	tableName string
+	bucket    *blob.Bucket
+}
+
+//go:embed schema.sql
+var tableSchema string
+
+// NewPostgresBucket creates a new Bucket.
+func NewPostgresBucket(ctx context.Context, connString string) (*Bucket, error) {
+	u, err := url.Parse(connString)
```

Comment:
OpenBucketURL takes a "*url.Url" that you call String on to call this which straight away parses back into a url. I think just change connString to be a *url.Url